src/fastoad/cmd/api.py

```diff
@@ -137,79 +139,169 @@ def generate_notebooks(
     return destination_path
 
 
-def generate_configuration_file(
-    configuration_file_path: str,
+def _generate_user_file(
+    user_file_type: str,
```

Comment:
I followed your suggestion of using an Enum class. I removed the FastUnknownUserFileTypeForGeneration exception since it is now naturally enforced by the Enum.